fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -395,11 +396,17 @@ public void dropBackend(String host, int heartbeatPort, boolean needCheckUnforce
             // remove worker
             if (Config.integrate_starmgr) {
                 long starletPort = droppedBackend.getStarletPort();
+                // only need remove worker when this be get dead after it became alive
+                // if it is dead all the time, not need to remove worker
+                boolean gotStarletPort = Deencapsulation.getField(droppedBackend, "gotStarletPort");
```

Comment:
what if the worker is added with wrong starlet port, and its heartbeat is never sent to FE/StarManager, will it be easy to remove this worker anymore?